udpa/type/v1/typed_struct.proto

```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
+
+package udpa.type.v1;
+
+option java_outer_classname = "TypedStructProto";
+option java_multiple_files = true;
+option java_package = "com.github.udpa.udpa.type.v1";
+
+import "validate/validate.proto";
+import "google/protobuf/struct.proto";
+
+// A TypedStruct contains an arbitrary JSON serialized protocol buffer message with a URL that
+// describes the type of the serialized message. This is very similar to google.protobuf.Any,
+// instead of having protocol buffer binary, this employs google.protobuf.Struct as value.
+//
+// This message is intended to be embedded inside Any, so it shouldn't be directly referred
+// from other UDPA messages.
+//
+// When packing an opaque extension config, packing the expected type into Any is preferred
+// a wherever possible for its efficiency. Pack TypedStruct inside Any only if:
```

Comment:
I would change the wording and say "TypedStruct should be used only if a proto descriptor is not available, for example if:".